gateway/src/main/java/org/georchestra/gateway/filter/headers/RemoveHeadersGatewayFilterFactory.java

```diff
@@ -78,14 +97,37 @@ public GatewayFilter apply(RegExConfig regexConfig) {
         return (exchange, chain) -> {
             final RegExConfig config = regexConfig;// == null ? DEFAULT_SECURITY_HEADERS_CONFIG : regexConfig;
             HttpHeaders incoming = exchange.getRequest().getHeaders();
-            if (config.anyMatches(incoming)) {
+
+            if (config.anyMatches(incoming) && (!configProps.isHeaderAuthentication()
+                    || !headerAuthenticated(exchange.getRequest().getRemoteAddress().getAddress().toString()))) {
```

Comment:
Tested on the target infrastructure, performing a remote-debugging: `getRemoteAddress()` returns the actual client issuing the request, which is provided as the http headers `x-forwarded-*` family, which we can't trust.

Reply:
this should return the internal IP address of the upstream proxy (apache mod_mellon in our case):

```
exchange.getRequest().getNativeRequest().connectionRemoteAddress()
```

Reply:
... but there is a mismatch between the service ip & the pods one in our k8s infrastructure, so I can't tell how the check can be reliable. Maybe it could make more sense to just drop this feature from the application, and use the k8s network policies instead ?